cmd/cmd.go

```diff
@@ -130,32 +138,86 @@ func CreateHandler(cmd *cobra.Command, args []string) error {
 
 			fi, err := os.Stat(path)
 			if errors.Is(err, os.ErrNotExist) && modelfile.Commands[i].Name == "model" {
+				request.From = modelfile.Commands[i].Args
 				continue
 			} else if err != nil {
 				return err
 			}
 
+			var files []string
 			if fi.IsDir() {
 				// this is likely a safetensors or pytorch directory
 				// TODO make this work w/ adapters
-				tempfile, err := tempZipFiles(path)
+				files, err = getFileList(path)
 				if err != nil {
 					return err
 				}
-				defer os.RemoveAll(tempfile)
-
-				path = tempfile
-			}
-
-			digest, err := createBlob(cmd, client, path, spinner)
-			if err != nil {
-				return err
+				var apiFiles []api.File
+				for _, file := range files {
+					relPath, err := filepath.Rel(path, file)
+					if err != nil {
+						return err
+					}
+
+					digest, err := createBlob(cmd, client, file, spinner)
+					if err != nil {
+						return err
+					}
+					apiFiles = append(apiFiles, api.File{Path: relPath, Digest: digest})
+				}
+				if modelfile.Commands[i].Name == "model" {
+					cfr := api.CreateFromRequest{Type: "safetensors"}
+					cfr.Files = apiFiles
+					request.From = cfr
+				} else {
+					request.Adapters = apiFiles
+				}
+			} else {
```

Comment:
Not sure if this makes it cleaner, but instead of the `if` here you could potentially `break` out of the switch since it's getting a bit pyramid-y

Reply:
I cut this down to put the part above into a separate function so it's less "pyramid-y". There's still a lot of conditional logic but it's much easier to read.